src/engine.cc

```diff
@@ -0,0 +1,83 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Promise::Resolver> resolver;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  HandleScope scope(isolate);
+  CallbackData *data = (CallbackData *)user_data;
+
+  GError *error = NULL;
+  EkncContentObjectModel *model;
+
+  Local<Promise::Resolver> resolver = Local<Promise::Resolver>::New(isolate, data->resolver);
```

Comment:
Oh, I misread it initially, you are creating a new `Local<>`, not a new `Promise::Resolver`. Still don't understand why you can't call methods on the `Persistent<>`, in SpiderMonkey that works fine, but if it doesn't work, it doesn't work...